### PR TITLE
add cloneable to Car and Engine

### PR DESCRIPTION
**Pull request by karsv**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+        this.height = height;
+        this.weight = weight;
+        this.color = color;
+        this.maxSpeed = maxSpeed;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
+        return new Car(this.height, this.weight, this.color, this.maxSpeed, this.engine);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Car.java]: let's remove word `this` everywhere

and what about `engine`? how will we create clone of engine?

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,14 @@
 package core.basesyntax;
 
-public class Engine {
+public class Engine implements Cloneable {
+    private int ne;
+    private int cylinders;
+    private int mk;
+    private int nd;
+    private int sd;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Engine.java]: name of this variables are very strange

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
 
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        return super.clone();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Engine.java]: try not use super.clone()

File: src/main/java/core/basesyntax/Engine.java (hunk 2 above)
Comment (karsv) [reply src/main/java/core/basesyntax/Engine.java]: theкe are the designations of real parameters. But ive already change it.

**Comment by Tarabarnik** [comment]
please, be careful to fix each comment or give a question to us

**Comment by karsv** [comment]
im really sorry! i dont know how explain this, but im secondary made changes at Engine

**Pull request closed without merging**